cylc/rose/utilities.py

```diff
@@ -869,6 +870,13 @@ def export_environment(environment: Dict[str, str]) -> None:
     for key, val in environment.items():
         os.environ[key] = val
 
+    # If env vars have been set we want to force reload
+    # the global config so that the value of this vars
+    # can be used by Jinja2 in the global config.
+    # https://github.com/cylc/cylc-rose/issues/237
+    if environment:
```

Comment:
Note, the environment is always populated as a result of automatic Rose variables.